### PR TITLE
Compat: disallow textureLoad with depth textures

### PR DESCRIPTION
**Pull request by greggman**

### CODE REVIEW

File: proposals/compatibility-mode.md [hunk 1]
```diff
@@ -180,6 +180,16 @@ create cube maps that are not exactly 6 layers.
 
 **Justification**: OpenGL ES 3.1 does not support bgra8unorm-srgb textures.
 
+## 16. Disallow `textureLoad` with `texture_depth?` textures
+
+If a `texture_depth`, `texture_depth_2d_array`, or `texture_depth_cube` are used via an entry point
```
Comment (kainino0x) [inline proposals/compatibility-mode.md:185]:
```suggestion
If a `texture_depth`, `texture_depth_2d_array`, or `texture_depth_cube` are used in a `textureLoad` call via an entry point
```

**Review by kainino0x — COMMENTED** [review]
fixes #4512?

File: proposals/compatibility-mode.md (hunk 1 above)
Comment (kainino0x) [inline proposals/compatibility-mode.md:185]: Question I guess, is it, any entry point in the module, or is it the one used to create the pipeline?

This is a general question though - I thought I knew the answer until recently, but then found out that WGSL went the other way for features like f16.

**Review by teoxoy — APPROVED** [review]

**Comment by greggman** [comment]
I'm going to merge this for now. We can adjust all issues related to https://github.com/gpuweb/gpuweb/issues/4568 depending on how that is decided

**Pull request merged**